src/bin/hexyl.rs

```diff
@@ -94,6 +129,19 @@ fn run() -> Result<(), Box<dyn std::error::Error>> {
         reader = Box::new(reader.take(length));
     }
 
+    let byterange_arg = matches.value_of("range");
+    let mut range_offset = 0;
+    if let Some(range) = byterange_arg {
+        if let Ok((offset, num_bytes)) = parse_range(range) {
+            range_offset = offset;
+            let mut discard = vec![0u8; offset as usize];
+            reader
+                .read_exact(&mut discard)
```

Comment:
Hm, that seems a bit hacky, but I also couldn't find a better solution after a quick search.

I'm not sure if we should leave it like this. If somebody wants to skip the first few MB or even GB of a file/input, we would actually allocate a buffer of this size on the heap which could easily fail. Even if it doesn't fail, it seems wrong to allocate a huge buffer for simply discarding everything inside it.

If that helps, I think we could actually use the `BufRead` trait instead of just `Read`. We would simply need to create a `BufReader` for the input file. The `StdinLock` already implements `BufRead`.

Reply:
What we want is `Seek`: https://doc.rust-lang.org/stable/std/io/struct.BufReader.html#impl-Seek

Reply:
I did a bit of research on this and Stdin doesn't appear to be seekable no matter what I do. I'll continue exploring other options for this.

Reply:
Maybe read to a 1024 bytes or so big slice multiple times as necessary and then read to a smaller slice as necessary.